src/npm-utils.js

```diff
@@ -131,14 +130,10 @@ module.exports = {
    * @return {Promise} The promise object.
    */
   downloadTarball: function(pkg, version, dir) {
-    var oldCWD = process.cwd();
     process.chdir(dir);
     return execPackCommand([pkg + '@' + version])
       .then(function(filename) {
         return path.resolve(dir || process.cwd(), filename[0]);
-      })
-      .finally(function() {
```

Comment:
Why did you remove this? To avoid side effect, I think it's cleaner to reset the current working directory to its original value, are you ok with that?